### PR TITLE
Allow to join a course without a passcode #1380

### PR DESCRIPTION
**Pull request by mauditecandela**
For those cases where we want to allow the users to join a program without having a passcode, we will display a different kind of button that doesn't require to fill out a passcode input.
This PR: 
- Adds a new method to the Course model that checks if the course requires a passcode and if this passcode is blank. Will return true only if the passcode IS blank and IS NOT required
- Adds a new functionality to available_actions displaying a different kind of button if the course allows registration without a passcode
- Includes a new method in the SelfEnrollment Controller that will allow the user indeed to register in this course they don't provide the passcode. 
- Adds an English translation for the Modal Prompt.

### CODE REVIEW

**Comment by ragesoss** [comment]
Did you get the i18n working?

This should have tests too. There are related tests *with* a passcode in `spec/features/student_role_spec.rb`.

**Comment by mauditecandela** [comment]
Yes, the i18n thing is working. 
I will work on the tests now.

**Comment by mauditecandela** [comment]
@ragesoss do you think writing a test in the same place (student_role_spec) is a good idea? I think it makes sense.

**Comment by ragesoss** [comment]
Yep!

**Comment by ragesoss** [comment]
@mauditecandela if you have the dev build of the assets, then that's normal; the error-level warnings for deprecated react features cause the capybara tests to fail. Do `gulp build` and those errors go away.

If you are using the production build (`gulp build`) and you have errors still, we should look for the cause.

File: spec/features/student_role_spec.rb [hunk 1]
```diff
@@ -24,6 +24,17 @@
...
+    create(:editathon,
+           title: 'An Example Editathon',
+           school: 'University',
+           term: 'Term',
+           slug: 'University/An_Example_Editathon_(Term)',
+           submitted: true,
+           passcode: '',
+           start: '2015-01-01'.to_date,
+           end: '2020-01-01'.to_date)
+  end
```
Comment (ragesoss) [inline spec/features/student_role_spec.rb]: Need to add this to a campaign as well.

**Pull request merged**